### PR TITLE
fix: rule.SetOnResponse = true not work

### PR DESCRIPTION
**Pull request by isayme**
`rule.SetOnResponse = true` is not working, because plugin current handle response headers before any response header written. I add some tests cases and fixed the problem.

### CODE REVIEW

**Comment by tomMoulard** [comment]
Hi @isayme,

Yeah you are right, that's a nice way to fix this issue!

Might fix #51 and #58

Could you please fix the linting?

**Comment by isayme** [comment]
@tomMoulard linting fiexd. BTW, #50 might also fixed.

File: htransformation.go [hunk 1]
```diff
@@ -59,22 +63,90 @@ func New(_ context.Context, next http.Handler, config *Config, name string) (htt
...
+func (wrw *wrappedResponseWriter) Header() http.Header {
+	return wrw.rw.Header()
+}
+
+func (wrw *wrappedResponseWriter) Write(p []byte) (int, error) {
+	wrw.handleResponseHeader()
+
+	n, err := wrw.rw.Write(p)
+	if err != nil {
+		return 0, fmt.Errorf("%w: write response", err)
```
Comment (tomMoulard) [inline htransformation.go:129]:
```suggestion
		return 0, fmt.Errorf("write response: %w", err)
```

File: htransformation.go [hunk 2]
```diff
@@ -59,22 +63,90 @@ func New(_ context.Context, next http.Handler, config *Config, name string) (htt
...
+
+func (wrw *wrappedResponseWriter) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+	hijacker, ok := wrw.rw.(http.Hijacker)
+	if !ok {
+		return nil, nil, types.ErrNotHTTPHijacker
+	}
+
+	conn, rw, err := hijacker.Hijack()
+	if err != nil {
+		return nil, nil, fmt.Errorf("%w: Hijack", err)
```
Comment (tomMoulard) [inline htransformation.go:148]:
```suggestion
		return nil, nil, fmt.Errorf("hijack: %w", err)
```

**Review by tomMoulard — APPROVED** [review]

**Pull request merged**